src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -760,32 +774,55 @@ private List<LaunchCommand> GetInitializeCommands()
                         commands.Add(new LaunchCommand("-environment-cd " + escapedDir));
                     }
 
-                    // TODO: The last clause for LLDB may need to be changed when we support LLDB on Linux
+                    // TODO: The last clause for LLDB may need to be changed when we support LLDB on Linux as LLDB's tty redirection doesn't work.
                     if (localLaunchOptions != null &&
-                        this.MICommandFactory.UseExternalConsoleForLocalLaunch(localLaunchOptions) &&
-                        (PlatformUtilities.IsWindows() || (PlatformUtilities.IsOSX() && this.MICommandFactory.Mode == MIMode.Lldb)))
+                        ((PlatformUtilities.IsWindows() && localLaunchOptions.UseExternalConsole)
+                        || (PlatformUtilities.IsOSX() && this.MICommandFactory.Mode == MIMode.Lldb)))
```

Comment:
This test is different than the test on line 197. Is that a problem? If so, I would suggest extract this into a helper (UseDebuggerNewConsoleOption) that can be called from both places (so the last line of 197 would be `&& !UseDebuggerNewConsoleOption(options)`).